portfolio/src/main/java/com/google/sps/servlets/Comments.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.servlets;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.KeyFactory;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.HashMap;
+
+public class Comments {
+
+  /**
+  * Gets comments from datastore at specified ID
+  */
+  public static String getComments(String collectionID) {
+    // Create datastore query
+    Query query = new Query(collectionID).addSort("timestamp", SortDirection.DESCENDING);
+
+    // Prepare database to be queried
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Get query results
+    PreparedQuery results = datastore.prepare(query);
+
+    Map<String, List<List<String>>> comments = new HashMap<>();
+    List<String> arrangedKeys = new ArrayList<>();
+
+    // Interate through entities in query results
+    for (Entity entity : results.asIterable()) {
+      String key;
+
+      if (entity.getParent() != null) {
+        key = KeyFactory.keyToString(entity.getParent());
+      } else {
+        key = KeyFactory.keyToString(entity.getKey());
+      }
+
+      // Retrieve comment inputs from entity
+      String name = (String) entity.getProperty("name");
+      String comment = (String) entity.getProperty("comment");
+      String timestamp = (String) entity.getProperty("timestamp");
+
+      // Store comment
+      if (!comments.containsKey(key)) {
+        comments.put(key, new ArrayList<List<String>>());
+        arrangedKeys.add(key);
+      }
+
+      comments.get(key).add(Arrays.asList(key, timestamp, name, comment));
```

Comment:
Instead of using a list could we create a comment object that has a key, timestamp, name, and comment?

Reply:
Yes. Will do.